app/src/main/java/ar/delellis/quicknotes/api/helper/NoteCompare.java

```diff
@@ -0,0 +1,58 @@
+package ar.delellis.quicknotes.api.helper;
+
+import com.google.gson.Gson;
+
+import ar.delellis.quicknotes.model.Note;
+
+/**
+ * Helper to compare different versions of notes for major changes
+ */
+public final class NoteCompare {
+
+    /**
+     * Compare specific fields from two Note elements.
+     * If there is any difference or fields are present in one,
+     * but not the other objects, this will return true.
+     *
+     * @param shadowCopy saved copy Note
+     * @param current    currently edited Note
+     * @return true if there is shadowCopy difference in the relevant fields
+     */
+    public static boolean compareNotes(Note shadowCopy, Note current) {
+
+        // Lets first check if one field is null and the other not (difference between shadowCopy and current)
+        // This is so we don't run into shadowCopy NPE later when accessing methods of fields.
+        if (shadowCopy.getTitle() == null ^ current.getTitle() == null
```

Comment:
"^" is binary XOR????????? :confused: :question: 
I truly believe that you are exaggerating here. But if you assure me that it is necessary, I can leave it and in any case change it later.

Reply:
Well like this we can easily compare if one of them differs (null-wise). If both are false, that's fine, if both are true, that's fine as well. That way its a little bit less code 👍🏼